typegen/src/tests/mod.rs

```diff
@@ -1697,3 +1697,67 @@ fn assoc_types_no_skip_params() {
 
     assert_eq!(dedup_code.to_string(), expected_code.to_string());
 }
+
+#[test]
+fn alloc_crate_path_replacement() {
+    use std::collections::BTreeMap;
+
+    #[derive(TypeInfo)]
+    #[allow(unused)]
+    pub struct Foo {
+        a: Vec<Foo>,
+        b: Box<u8>,
+        c: String,
+        e: BTreeMap<String, Box<u8>>,
+    }
+
+    let std_code = Testgen::new()
+        .with::<Foo>()
+        .try_gen_tests_mod(Default::default(), false)
+        .unwrap();
+
+    let no_std_code = Testgen::new()
+        .with::<Foo>()
+        .try_gen_tests_mod(
+            TypeGeneratorSettings {
+                alloc_crate_path: AllocCratePath::Custom(parse_quote!(::subxt_core::alloc)),
+                ..Default::default()
+            },
+            false,
+        )
+        .unwrap();
+
+    let expected_std_code = quote! {
+        pub mod tests {
+            use super::types;
+            pub struct Foo {
+                pub a: ::std::vec::Vec<types::scale_typegen::tests::Foo>,
+                pub b: ::std::boxed::Box<::core::primitive::u8>,
+                pub c: ::std::string::String,
+                pub e: ::std::boxed::Box<
+                    ::std::collections::BTreeMap<::std::string::String, ::core::primitive::u8>
+                >,
+            }
+        }
+    };
+
+    let expected_no_std_code = quote! {
+        pub mod tests {
+            use super::types;
+            pub struct Foo {
+                pub a: ::subxt_core::alloc::vec::Vec<types::scale_typegen::tests::Foo>,
```

Comment:
Should we document the types and paths of those types that we'd expect?

For example, to say something like "We'd expect the Custom path to export `vec::Vec`, `string::String`, `boxed::Box`", don't think that's a big deal since user's will probably get a nice error back for those

Reply:
I think that the user should be expected to point to something exporting all of the types from `alloc`.